### ISSUE TITLE
Commit to GH from JS

### ISSUE THREAD

**Issue by DominiqueMakowski** [RealityBending/TemplateOnlineExperiment#1]
let's open an issue to track my attempts ^^ thanks a ton @penguimelia for having looked into that 🙌 ☺️

**Comment by DominiqueMakowski** [comment]
When I add `<script type="module" src="experiment.js"></script>` locally (i.e. when I open the html in my browser) I get the following

```js
Access to script at 'file:///D:/Dropbox/RealityBendingLab/TemplateOnlineExperiment/experiment.js' from origin 'null' has been blocked by CORS policy: Cross origin requests are only supported for protocol schemes: http, data, chrome, chrome-extension, chrome-untrusted, https.
experiment.js:1 Failed to load resource: net::ERR_FAILED
```

**Comment by DominiqueMakowski** [comment]
Note that this is me with JS: 

![](https://lh3.googleusercontent.com/pw/ACtC-3dLNwkcd43fqxQ5JT3NAIZoz_Hlj4OmNFkqVSjM9GjOw6tBrk65tlVVNVQ4XBc95uZLl4y47PD6A6uIglpkvgMcrhyoa1Q8W079cbUnqS1PDhLsK8EYP3ANmWnjG6SAs62AET2BxmqKiR4jO9RbXGrqcg=w600-h370-no?authuser=0)

**Comment by penguimelia** [comment]
yeah, i mentioned to zen that this method will require setting up a local
web server. are you using VS Code? if you are, there’s a local server
extension which is super easy to use. otherwise you can google “local web
server setup” for something you can do on your machine too.

basically bc the site is running static on your machine, with no “web”
server infrastructure, for security reasons chrome prevents the local app
from accessing the web and vice versa (that’s probably a shitty explanation
lmao u can google CORS policy for a proper description 😂)

am out rn so i’m just replying on my phone — can def look at the Issues
later!!!

**Comment by DominiqueMakowski** [comment]
> basically bc the site is running static on your machine, with no “web”
server infrastructure, for security reasons chrome prevents the local app
from accessing the web and vice versa

That makes sense! I am using vscode, I'll give it a look thanks

It seems like importing the script as a module enforces the need to declare all variables is that possible? Previously the script worked but after that it errored saying that some variables were not explicitly declared   

> am out rn so i’m just replying on my phone — can def look at the Issues later!!!

no worries it's CNY afterall ☺️

**Comment by DominiqueMakowski** [comment]
> It seems like importing the script as a module enforces the need to declare all variables is that possible? Previously the script worked but after that it errored saying that some variables were not explicitly declared

Oh yea, I didn't notice you did that in your fork as well (I'm trying to redo and understand step by step :)

**Comment by DominiqueMakowski** [comment]
I have 
```js
Failed to load resource: the server responded with a status of 401 ()
request.js:65 Uncaught (in promise) HttpError: Bad credentials
    at https://cdn.skypack.dev/-/@octokit/request@v5.4.14-2bZGuWjihBc50hc0wwRm/dist=es2020,mode=imports/optimized/@octokit/request.js:65:23
```

I suspect it could be the Oauth token. Here's what I did to get it, I'm not sure it's the right way: 

- Went to my (personal) GH profile and to Settings
- Went to "Developper setttings" tab then to OAuth apps
- Created a new OAuth app (named jsPsychTemplate), then generated a new "Clien secret" and copy pasted the string of characters

**Comment by penguimelia** [comment]
yes that’s the right steps!! sorry was at f45. you also need to change the
name of the git repo owner etc as well! ah you didn’t need to create a new
oauth app, generating a new personal access token would have worked. if the
oauth app doesn’t work, then try using the personal access token instead!

also this won’t be allowed to live as plaintext in the repo as git will
invalidate that token, so we have to look into using the repo’s environment
secrets as well

**Comment by DominiqueMakowski** [comment]
> if the oauth app doesn’t work, then try using the personal access token instead!

Yeah it doesn't, I'm not sure what I did wrongly 😕 


![image](https://user-images.githubusercontent.com/8875533/107597443-9d2e2880-6c55-11eb-89d8-02ae3d6f7c8b.png)

I was the charcter string circled in red that I pasted here

https://github.com/RealityBending/TemplateOnlineExperiment/blob/cd7dd60cb3b67310682222f8570b458730aaa21e/experiment.js#L7

> also this won’t be allowed to live as plaintext in the repo as git will invalidate that token, so we have to look into using the repo’s environment secrets as well

I have no idea what it means, but I believe you 😅

**Comment by penguimelia** [comment]
i think oauth apps are used to connect other apps to your application, so maybe it doesn't work like that? try doing the personal access token one instead! 

> I have no idea what it means, but I believe you 😅

basically just pasting your personal access token in a repo like this is dangerous becauswe people can just come in and take it and mess up your stuff, so we have to secretly save it in the repository settings in a way that only github can see, if that makes more sense?

**Comment by penguimelia** [comment]
if it'll help, i can set up a zoom pairing session and we can go through that together!! might have to be tomorrow though

**Comment by DominiqueMakowski** [comment]
So I created a personal token access, and then I created an org secret (so that it can be reused in other future repos from the org?):

![image](https://user-images.githubusercontent.com/8875533/107598264-39f1c580-6c58-11eb-98a8-0375b70b67be.png)


with its value, but not sure now where to add that secret inside the code?

> i can set up a zoom pairing session and we can go through that together!! might have to be tomorrow though

Sure! whenever you prefer, I'm quite free since @zen-juen bo'joed me from CNY

**Comment by penguimelia** [comment]
hmmm brb let me try something out

**Comment by penguimelia** [comment]
so it looks like this is gonna be kinda complicated to set up as it involves some CI/CD stuff for github to be able to read the environment secret............ we'd need to create a Github Action to deploy the app with the environment variable but i have no idea how to do that either 😂

**Comment by penguimelia** [comment]
so am currently looking at this https://itsopensource.com/publish-github-pages-with-github-actions/
seems a little uhhh over the top but we won't need some of the steps i think. i have to work today but tomorrow i can tinker!! and we can zoom if you'd like :)

**Comment by DominiqueMakowski** [comment]
That works for me 👌 thanks again!

**Comment by DominiqueMakowski** [comment]
Though here https://paul.kinlan.me/creating-a-commit-with-multiple-files-to-github-with-js-on-the-web/ it seems like they somehow manage to do it? Or is it something different?

**Comment by penguimelia** [comment]
they’re using Firebase Auth, which i think is another JS library that
provides sign in UI elements out of the box. meaning it might create a pop
up for u to sign in to github when you start up the app

we can look into using that too!! might have issues with the importing of
it though but i haven’t looked too much into the repo yet

**Comment by DominiqueMakowski** [comment]
Oh I see, nah I'd prefer something that doesn't pop up anything and just commit silently behind the scenes

**Comment by zen-juen** [comment]
Thanks bae @penguimelia 😉 

This repo seems quite appropriate for loading secrets in github actions: https://secrethub.io/docs/guides/github-actions/ we might just be able to follow these steps directly? though for this the secret credential should be the secret of the repository rather than the organization! Not sure which is best but with my (very limited lolol) knowledge on this, the link here seems quite straightforward to follow

**Comment by zen-juen** [comment]
I also can't see the settings button for this repo :))))) @DominiqueMakowski

**Comment by penguimelia** [comment]
that tutorial is a little bit more complicated than what we'd actually need i thiiink. it's also set up for docker which is a whole other world we don't wanna explore 😅
this one is really the most straightforward one i've seen: https://itsopensource.com/publish-github-pages-with-github-actions/ we just need to remove the NPM steps and i thiiiink we should be good... but not sure i'd have to play around to see if it'll work lol

**Comment by penguimelia** [comment]
i also love how indistinguishable our profile icons are 😂 the colour schemes are the SAME

**Comment by DominiqueMakowski** [comment]
Lemme add you girls to this repo so u have full access

**Comment by DominiqueMakowski** [comment]
@penguimelia also, I'd like to avoid the commits to be tagged with my personal account ('cause it will mess up my activity and my send tons of notifications to the people that follow me ^^), what do you think could be the alternatives? I thought about creating a phantom GH account just for that, but I'm worried that GH will take it down if they see that it has only suspect activity. Maybe creating some sort of "bot"? Or maybe I can try again with the OAuth app? Unfortunately it seems like there is no "personal token" for the organization so that it could be committed via the org 😞 

![image](https://user-images.githubusercontent.com/8875533/108293899-c60f6a00-71cf-11eb-89b4-d0da358b74e4.png)
